libtransmission/span.h

```diff
@@ -0,0 +1,56 @@
+/*
+* This file Copyright (C) 2008-2014 Mnemosyne LLC
+*
+* It may be used under the GNU GPL versions 2 or 3
+* or any future license endorsed by Mnemosyne LLC.
+*
+*/
+
+#pragma once
+
+#ifndef __TRANSMISSION__
+#error only libtransmission should #include this header.
+#endif
+
+#include <array>
+#include <vector>
+
+#include "transmission.h"
+
+/// @brief Readonly non-owning view into a provided memory block with start pointer and size.
+/// In C++20 this appears in standard library as std::span and remotely similar usage.
+template <typename T>
+class Span
+{
+public:
+    Span(const T* ptr, size_t size)
+        : ptr_{ ptr }
+        , size_{ size }
+    {
+    }
+
+    const T* begin() const
+    {
+        return this->ptr_;
+    }
+
+    const T* end() const
+    {
+        return this->ptr_ + this->size_;
+    }
+
+private:
+    const T* ptr_;
+    size_t size_;
+
+public:
+    const T* getPtr() const
+    {
+        return ptr_;
+    }
+
+    [[nodiscard]] size_t getSize() const
+    {
+        return size_;
+    }
```

Comment:
the public accessors should go before the private fields

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rl-order